src/MetaModels/Attribute/Select/Select.php

```diff
@@ -102,6 +103,10 @@ public function getAttributeSettingNames()
      */
     public function valueToWidget($varValue)
     {
+        if ($this->isTreePicker()) {
+            return [$varValue[$this->getIdColumn()]];
```

Comment:
Put the convert in a method and set the method as deprecated in 2.0 and where removed in 2.2.
And trigger a error notice.